bellows/uart.py

```diff
@@ -125,7 +125,9 @@ def rstack_frame_received(self, data):
             LOGGER.warn("Reset future is None")
             return
 
-        self._reset_future.set_result(True)
+        # Make sure that the reset_future is not done
+        if self._reset_future.done() is not True:
```

Comment:
`if not self._reset_future.done()` ?

Reply:
It was just a easy way, not to rewrite the test cases. But I can just rewrite them to make them pass.